lib/json-schema.ts

```diff
@@ -0,0 +1,364 @@
+import AJV from 'ajv';
+import addFormats from 'ajv-formats';
+import type { JSONSchema6 } from 'json-schema';
+
+function isString(value: unknown): value is string {
+	return typeof value === 'string' || value instanceof String;
+}
+// Adds extra formats to an AJV instance
+const configureAjv = (ajv: AJV) => {
+	ajv.addFormat('markdown', isString);
+	ajv.addFormat('mermaid', isString);
+
+	// Add additional formats
+	addFormats(ajv);
+};
+
+const JSON_SCHEMA_SCHEMA = {
+	$schema: 'http://json-schema.org/draft-07/schema#',
+	$id: 'jellyfish-meta-schema',
```

Comment:
I think this schema can be simplified, but I'm not familiar with json-schema or AJV to identify what is superfluous